### PR TITLE
Lightning-Address (sending)

### PR DESCRIPTION
**Pull request by nochiel**
This depends on https://github.com/breez/breezmobile/pull/560.

### CODE REVIEW

File: account/lnurl.go [hunk 1]
```diff
@@ -29,14 +30,60 @@ type LoginResponse struct {
...
+func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err error) {
+	var iparams lnurl.LNURLParams
+	var rawurl, lightningAddress string
+
 	encodedLnurl, ok := lnurl.FindLNURLInText(rawString)
-	if !ok {
-		return nil, fmt.Errorf("'%s' does not contain an LNURL.", rawString)
+	if ok {
+		a.log.Infof("HandleLNURL %v", encodedLnurl)
+		rawurl, iparams, err = lnurl.HandleLNURL(encodedLnurl)
```
Comment (roeierez) [inline account/lnurl.go:40]: In general in go it is a convention to return error immediately when we have one as it makes a more clear flow.
I think here we should check for error and return error immediately

File: account/lnurl.go [hunk 2]
```diff
@@ -29,14 +30,60 @@ type LoginResponse struct {
...
+
+	} else {
+
+		err = fmt.Errorf("%q does not contain an LNURL.", rawString)
+
+		if strings.Contains(rawString, "@") {
+			// ref. Lightning-Address https://github.com/andrerfneves/lightning-address/blob/master/DIY.md
+
+			err = nil
+			parts := strings.Split(rawString, "@")
```
Comment (roeierez) [inline account/lnurl.go:50]: Can we use the standard function https://pkg.go.dev/net/mail#ParseAddress instead

File: account/lnurl.go [hunk 3]
```diff
@@ -29,14 +30,60 @@ type LoginResponse struct {
...
+				err = fmt.Errorf("%q is not an e-mail address", rawString)
+			} else {
+
+				url := &url.URL{
+					Scheme: "https",
+					Host:   parts[1],
+					Path:   ".well-known/lnurlp/" + parts[0],
+				}
+
+				resp, err := http.Get(url.String())
```
Comment (roeierez) [inline account/lnurl.go:61]: Did you try calling `lnurl. HandleLNURL` here? This function has the ability to handle 'https' prefixed urls and perform the get request.

**Review by roeierez — CHANGES_REQUESTED** [review]
Thanks for the PR Nick.

File: account/lnurl.go (hunk 2 above)
Comment (nochiel) [reply account/lnurl.go:50]: Yes, to validate the address. But, as far as I can tell, it doesn't do the `username`, `host` splitting that we need to construct the lnurl.

File: account/lnurl.go [hunk 4]
```diff
@@ -29,18 +31,47 @@ type LoginResponse struct {
...
+	var iparams lnurl.LNURLParams
+	var rawurl, lightningAddress string
+
+	handleLNUrlError := fmt.Errorf("%q does not contain an LNURL.", rawString)
+
 	encodedLnurl, ok := lnurl.FindLNURLInText(rawString)
-	if !ok {
-		return nil, fmt.Errorf("'%s' does not contain an LNURL.", rawString)
-	}
+	if ok {
```
Comment (roeierez) [inline account/lnurl.go:42]: When the rawString is an email address the only difference is an additional step to construct the lnurl URL.
Why not use this step only when the `FindLNURLInText` fails and then both flows merge to the same point at `lnurl.HandleLNURL`
We can also extract the url construction logic to its own function.

Something along the lines of:

```
if !ok {
  encodedURL, err := lnurlFromEmail(rawString)
  if err != nil {
    return nil, err
  }
}

rawurl, iparams, err := lnurl.HandleLNURL(encodedLnurl)
```

**Review by roeierez — CHANGES_REQUESTED** [review]

**Review by roeierez — APPROVED** [review]

**Pull request merged**